core/services/major_tom/src/core.py

```diff
@@ -0,0 +1,149 @@
+import gzip
+import http
+import json
+import os
+import shutil
+import time
+from dataclasses import asdict, dataclass
+from typing import Any, Callable, Dict, List
+
+import loguru
+import requests
+import speedtest
+from loguru import logger
+
+
+def formatter(record: "loguru.Record") -> str:
+    # Note this function returns the string to be formatted, not the actual message to be logged
+    record["extra"]["serialized"] = json.dumps(record["message"])
+    return "{extra[serialized]}\n"
+
+
+def is_online() -> bool:
+    return get_latency() > 0
+
+
+def get_latency() -> float:
+    try:
+        servers: List[str] = []
+        st = speedtest.Speedtest()
+        st.get_servers(servers)
+        best_server = st.get_best_server()
+        ping = best_server["latency"]
+        return float(ping)
+    except Exception:
+        return -1.0
+
+
+@dataclass
+class DefaultPayload:
+    log_session_uuid: str
+    order: int
+    timestamp: str
+    hardware_id: str
+    blueos_id: str
+    data: dict[str, Any]
```

Comment:
Since this is a generic dict, aren't we supposed to have the serialized data with a defined type property to specify what is exactly in this field ?

Like:
```
data: {
    type: 'AnonymousTelemetryRecord'
    ...
}
````